smslibrary/src/main/java/com/eis/smslibrary/listeners/SMSReceivedServiceListener.java

```diff
@@ -0,0 +1,47 @@
+package com.eis.smslibrary.listeners;
+
+import android.app.IntentService;
+import android.content.Intent;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.eis.smslibrary.SMSMessage;
+import com.eis.smslibrary.SMSReceivedBroadcastReceiver;
+
+/**
+ * Abstract class to implement in order to wake up the service when a message is received
+ *
+ * @author Luca Crema, Alberto Ursino, Marco Tommasini, Alessandra Tonin
+ * @since 29/11/2019
+ */
+public abstract class SMSReceivedServiceListener extends IntentService {
+
+    /**
+     * Creates an IntentService. Invoked by your subclass's constructor.
+     *
+     * @param name Used to name the worker thread, important only for debugging.
+     */
+    protected SMSReceivedServiceListener(@NonNull String name) {
+        super(name);
+    }
+
+    /**
+     * Callback for when a sms message sent from this library is received
+     *
+     * @param message the received message
+     */
+    public abstract void onMessageReceived(SMSMessage message);
+
+    /**
+     * Handles the service call, extracts the message from the intent extras and calls {@link #onMessageReceived(SMSMessage)}
+     *
+     * @param intent intent passed from the broadcastReceiver
```

Comment:
Repetition

Reply:
Honestly I don't think being so picky about param documentation "repetitions" is so necessary. I remember you that in the documentation it becomes `@param intent - intent passed from the broadcastReceiver` which in my opinion is better than `@param intent - passed from the broadcastReceiver`. I would even go as far as writing it as `@param intent - The intent passed from the broadcastReceiver`

Reply:
I did not know this fact of documentation.
Thanks to having pointed this out to me, now I feel even more picky.
However, I welcome your correction: `@param intent - The intent passed from the broadcastReceiver`

Reply:
I was "picky" too, preferring the last solution, but since I found out (thanks to @CremaLuca ) that even in the official android api it would be sometimes written as `@param intent - passed from the broadcastReceiver` That I've just learned to accept all 3 solutions, (even though I'll still write the last one 😄 )